kerby-kerb/kerb-client/src/main/java/org/apache/kerby/kerberos/kerb/client/KrbClientBase.java

```diff
@@ -271,15 +271,23 @@ public void storeTicket(TgtTicket tgtTicket,
      */
     public void storeTicket(SgtTicket sgtTicket, File ccacheFile) throws KrbException {
         LOG.info("Storing the sgt to the credential cache file.");
-        if (!ccacheFile.exists()) {
+        boolean isFreshNew = !ccacheFile.exists();
+        if (isFreshNew) {
             createCacheFile(ccacheFile);
         }
         if (ccacheFile.exists() && ccacheFile.canWrite()) {
-            CredentialCache cCache = new CredentialCache();
+            
             try {
-                cCache.load(ccacheFile);
-                cCache.addCredential(new Credential(sgtTicket, sgtTicket.getClientPrincipal()));
-                cCache.setPrimaryPrincipal(sgtTicket.getClientPrincipal());
+                CredentialCache cCache;
+                if(!isFreshNew){
+                    //Remind, contructor sets the cCache client principal from the sgtTicket one
+                    cCache = new CredentialCache(sgtTicket); 
```

Comment:
Why not "new CredentialCache()" here as the code was before?

Reply:
right..

Reply:
I should also remove, in the same block, the cCache.setPrimaryPrincipal(sgtTicket.getClientPrincipal()); because adding an SGT should not change the primary client principal.. I think